mods/d2k/rules/misc.yaml

```diff
@@ -152,3 +152,97 @@ wormspawner:
 	RenderEditorOnly:
 	BodyOrientation:
 	WormSpawner:
+
+upgrade.conyard:
+	Tooltip:
+		Name: Construction Yard Upgrade
+		Description: Upgrade for the Construction Yard
+	Buildable:
+		BuildPaletteOrder: 50
+		Prerequisites: conyard
+		Queue: Upgrade
+		BuildLimit: 1
+	Valued:
+		Cost: 1000
+	RenderSprites:
+		Image: conyard.harkonnen
+		RaceImages:
+			atreides: conyard.atreides
+			ordos: conyard.ordos
+			corrino: conyard.corrino
+
+upgrade.barracks:
+	Tooltip:
+		Name: Barracks Upgrade
+		Description: Upgrade for the Barracks
+	Buildable:
+		BuildPaletteOrder: 50
+		Prerequisites: barracks
+		Queue: Upgrade
+		BuildLimit: 1
+	Valued:
+		Cost: 500
+	RenderSprites:
+		Image: barracks.harkonnen
+		RaceImages:
+			atreides: barracks.atreides
+			ordos: barracks.ordos
+
+upgrade.light:
+	Tooltip:
+		Name: Light Factory Upgrade
+		Description: Upgrade for the Light Factory
+	Buildable:
+		BuildPaletteOrder: 50
+		Prerequisites: light
+		Queue: Upgrade
+		BuildLimit: 1
+	Valued:
+		Cost: 400
+	RenderSprites:
+		Image: light.harkonnen
+		RaceImages:
+			atreides: light.atreides
+			ordos: light.ordos
+
+upgrade.heavy:
+	Tooltip:
+		Name: Heavy Factory Upgrade
+		Description: Upgrade for the Heavy Factory
+	Buildable:
+		BuildPaletteOrder: 50
+		Prerequisites: heavy
+		Queue: Upgrade
+		BuildLimit: 1
+	Valued:
+		Cost: 800
+	RenderSprites:
+		Image: heavy.harkonnen
+		RaceImages:
+			atreides: heavy.atreides
+			ordos: heavy.ordos
+			corrino: heavy.corrino
+
+upgrade.hightech:
+	Tooltip:
+		Name: High Tech Factory Upgrade
+		Description: Upgrade for the High Tech Factory
+	Buildable:
+		BuildPaletteOrder: 50
+		Prerequisites: ~light.at_upg, ~techlevel.superweapons
+		Queue: Upgrade
+		BuildLimit: 1
+	Valued:
+		Cost: 1500
+	AirstrikePower:
```

Comment:
Putting this here means that you'll always be able to use it, even if you lose your tech center.  You should instead keep it on hightech, but define `Prerequisites: upgrade.hightech` on the power trait.